### PR TITLE
make-register-form

### PR DESCRIPTION
**Pull request by Carolinecrepin**

### CODE REVIEW

File: src/Entity/User.php [hunk 1]
```diff
@@ -4,11 +4,13 @@
 
 use App\Repository\UserRepository;
 use Doctrine\ORM\Mapping as ORM;
+use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
 use Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface;
 use Symfony\Component\Security\Core\User\UserInterface;
 
 /**
  * @ORM\Entity(repositoryClass=UserRepository::class)
+ * @UniqueEntity(fields={"email"}, message="There is already an account with this email")
```
Comment (elodiedaubie) [inline src/Entity/User.php]: Hello, this message should be in french

File: templates/login/index.html.twig [hunk 2]
```diff
@@ -29,7 +29,7 @@
                 <a href="#"><p>Mot de passe oublié ?</p></a>
             </div>
             <div class="not-account">
-                <a href="#"><p>Vous n'avez pas de compte</p></a>
+                <a href=""><p>Vous n'avez pas de compte</p></a>
```
Comment (elodiedaubie) [inline templates/login/index.html.twig]: I think the link to register page is missing

File: templates/registration/confirmation_email.html.twig [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<h1>Hi! Please confirm your email!</h1>
+
+<p>
+    Please confirm your email address by clicking the following link: <br><br>
```
Comment (elodiedaubie) [inline templates/registration/confirmation_email.html.twig]: Les textes de ces pages devraient être en français

File: templates/registration/register.html.twig [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{% extends 'base.html.twig' %}
+
+{% block title %}Register{% endblock %}
```
Comment (elodiedaubie) [inline templates/registration/register.html.twig]: Le titre devrait être en français

File: templates/registration/register.html.twig [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    {{ form_start(registrationForm) }}
+        {{ form_row(registrationForm.email) }}
+        {{ form_row(registrationForm.plainPassword, {
+            label: 'Password'
+        }) }}
+        {{ form_row(registrationForm.agreeTerms) }}
+
+        <button type="submit" class="btn">Register</button>
+    {{ form_end(registrationForm) }}
+{% endblock %}
```
Comment (elodiedaubie) [inline templates/registration/register.html.twig:74]: Les termes du formulaire devraient être en français

**Review by elodiedaubie — CHANGES_REQUESTED** [review]
Hello! I put some comments on files directly. The form doesn't work because some properties of User Entity are not nullable (such as firstanme, lastname, phone number ...), we should discuss about that

**Review by elodiedaubie — CHANGES_REQUESTED** [review]
Thanks for the corrections. As we discussed, we need to add the mandatories fields (User properties) to the form

File: src/Entity/User.php [hunk 6]
```diff
@@ -35,28 +42,50 @@ class User implements UserInterface, PasswordAuthenticatedUserInterface
      */
     private string $password;
```
Comment (elodiedaubie) [inline src/Entity/User.php:43]: No assert here as we told, it will be in another US

File: templates/login/index.html.twig [hunk 7]
```diff
@@ -9,6 +9,13 @@
         <div class="img-user">
             <img src="{{ asset('build/images/user.png') }}" alt="img profil">
         </div>
+        {% for label, messages in app.flashes %}
```
Comment (elodiedaubie) [inline templates/login/index.html.twig]: I think that the flashes error message display should be in the base.html.twig, bellow the header

File: templates/registration/register.html.twig [hunk 8]
```diff
@@ -0,0 +1,77 @@
+{% extends 'base.html.twig' %}
+
+{% block title %}Inscription{% endblock %}
+
+{% block body %}
+    {% include "components/_header.html.twig" %}
+    {% for flash_error in app.flashes('verify_email_error') %}
```
Comment (elodiedaubie) [inline templates/registration/register.html.twig]: same here, I think that the flashes error message display should be in the base.html.twig, bellow the header

**Review by elodiedaubie — CHANGES_REQUESTED** [review]
Hello there, great work! I required only a  change about flash message

**Review by elodiedaubie — APPROVED** [review]

**Review by RomainClair — APPROVED** [review]

**Pull request merged**